cartoframes/context.py

```diff
@@ -663,6 +776,114 @@ def sync(self, dataframe, table_name):
         """
         pass
 
+    # TODO: replace query with this method
+    def querycopy(self, query, table_name=None, decode_geom=False):
+        """Pull the result from an arbitrary SQL query from a CARTO account
+        into a pandas DataFrame. Can also be used to perform database
+        operations (creating/dropping tables, adding columns, updates, etc.).
+
+        Args:
+            query (str): Query to run against CARTO user database. This data
+              will then be converted into a pandas DataFrame.
+            table_name (str, optional): If set, this will create a new
+              table in the user's CARTO account that is the result of the
+              query. Defaults to None (no table created).
+            decode_geom (bool, optional): Decodes CARTO's geometries into a
+              `Shapely <https://github.com/Toblerity/Shapely>`__
+              object that can be used, for example, in `GeoPandas
+              <http://geopandas.org/>`__.
+
+        Returns:
+            pandas.DataFrame: DataFrame representation of query supplied.
+            Pandas data types are inferred from PostgreSQL data types.
+            In the case of PostgreSQL date types, dates are attempted to be
+            converted, but on failure a data type 'object' is used.
+        """
+        self._debug_print(query=query)
+
+        # create table from query if requested
+        # TODO: turn this into a _table_touch with a custom query
+        if table_name:
+            # TODO: replace the following error catching with Import API
+            #  checking once Import API sql/table_name collision_strategy=skip
+            #  bug is fixed ref: support/1127
+            try:
+                # error if table exists, otherwise 'do nothing' (create/drop)
+                self.sql_client.send('''
+                    CREATE TABLE {0} AS SELECT 1;
```

Comment:
I understand this is TBD

I'd split into 2 distinct methods the case `if table_name` and the other.

Reply:
Good idea. There's a placeholder function in this PR called `_table_touch` that I intend to use for creating a new table from a query using the Import API, so I can swap out all of this code.

By the way, initial tests of that function yield some slowish performance: https://github.com/CartoDB/cartodb/issues/14055